src/System.Management.Automation/engine/runtime/Binding/Binders.cs

```diff
@@ -3946,6 +3946,11 @@ public override DynamicMetaObject FallbackGetIndex(DynamicMetaObject target, Dyn
                 return InvokeIndexer(target, indexes, errorSuggestion, defaultMember.MemberName).WriteToDebugLog(this);
             }
 
+            if (typeof(ITuple).IsAssignableFrom(target.LimitType))
```

Comment:
This doesn't feel like the right fix. `ITuple` specifies a `DefaultMemberAttribute`, but we don't find it above because `GetCustomAttributes` is only checking classes, not implemented interfaces.

The correct fix is to search interfaces for `DefaultMemberAttribute` - there are other interfaces that specify `DefaultMemberAttribute`, and some of those are not currently handled correctly:

```
System.Collections.IDictionary
System.Collections.IList
System.Collections.Generic.IDictionary`2
System.Collections.Generic.IList`1
System.Collections.Generic.IReadOnlyDictionary`2
System.Collections.Generic.IReadOnlyList`1
System.Runtime.CompilerServices.ITuple
System.Runtime.CompilerServices.IRuntimeVariables
System.Collections.Specialized.IOrderedDictionary
System.Linq.ILookup`2
System.Data.IColumnMappingCollection
System.Data.IDataParameterCollection
System.Data.IDataRecord
System.Data.ITableMappingCollection
System.ComponentModel.IDataErrorInfo
Newtonsoft.Json.Linq.IJEnumerable`1
```

Reply:
@lzybkr Good call! I had checked that none of the default `ITuple` implementations had that attribute but didn't think to check the interface itself.

I've updated the change to look for `DefaultMemberAttribute` in the loop that looks for `IDictionary<,>`.  I tested the update with a custom `ILookup<,>` with an explicitly implemented indexer, that didn't work previously but does now.

Reply:
Do now we support all these interfaces? If so, it's good to have a comment in the code and tests (or even extend the tests)
Also we need to document this.

Reply:
@iSazonov They were already supported.  What *wasn't* supported was indexing into classes that *explicitly* implement the indexer of an interface and define no other indexer.

All `ITuple` implementations explicitly implement the `ITuple` indexer, so while indexing tuples was technically supported, it wasn't working.

Reply:
@SeeminglyScience Sorry, I did not ask the question correctly. Your explanation is useful and I just believe that it will make it easier to understand the binder if it's in the code.

Reply:
@iSazonov Ah yes absolutely, I'll add that tonight.

Reply:
@iSazonov I've added a comment explaining purpose of the extra check in the loop.